### PR TITLE
OCI ATP Integration Part 1

### PR DESCRIPTION
**Pull request by arjav-desai**
This PR is initial work where calling api will generate the wallet and download it locally on end user's machine.

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -882,11 +882,6 @@
                 <artifactId>helidon-integrations-cdi-oci-objectstorage</artifactId>
                 <version>${helidon.version}</version>
             </dependency>
```
Comment (tomas-langer) [inline bom/pom.xml:884]: Changes in this file look wrong.

File: examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import io.helidon.integrations.oci.atp.GenerateAutonomousDatabaseWallet;
+import io.helidon.integrations.oci.atp.GenerateAutonomousDatabaseWalletRx;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * JAX-RS resource - REST API for the atp example.
+ */
+@Path("/atp")
+public class ATPResource {
```
Comment (tomas-langer) [inline examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java:47]: Naming in Helidon is strictly "camel case" - so this class should be named `AtpResource` (this is valid for any class, field, method or variable)

File: examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    private final OciAutonomousDB autonomousDB;
+
+    @Inject
+    ATPResource(OciAutonomousDB autonomousDB) {
```
Comment (tomas-langer) [inline examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java:51]: `OciAutonomousDB` should be `OciAutonomousDb`

**Review by tomas-langer — CHANGES_REQUESTED** [review]
Please update naming. Otherwise looks ok

File: bom/pom.xml (hunk 1 above)
Comment (arjav-desai) [reply bom/pom.xml:884]: Sorry, which change? We are making atp module available to both se/mp bundle, just like objectstorage/vault/helath etc.

Comment (arjav-desai) [reply bom/pom.xml:906]: Also, deps were arranged alphabetically.

File: examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        GenerateAutonomousDatabaseWallet.Response response = entity.get();
+
+        StreamingOutput stream = output -> response.writeTo(Channels.newChannel(output));
+
+        Response.ResponseBuilder ok = Response.ok(stream, MediaType.APPLICATION_OCTET_STREAM_TYPE)
+                .header("opc-request-id", ociResponse.headers().first("opc-request-id").orElse(""))
+                .header("request-id", ociResponse.requestId());
+        //.header(Http.Header.CONTENT_DISPOSITION, "attachment; filename=\"" + fileName + "\"")
```
[the remark this answers is not in the document]
Comment (arjav-desai) [reply examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java:77]: Removed it.

File: integrations/oci/atp/src/main/java/io/helidon/integrations/oci/atp/GenerateAutonomousDatabaseWalletRx.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            return new Request();
+        }
+
+        /**
+         * Set explicit password to encrypt the keys inside the wallet.
+         *
+         * @param walletPassword walletPassword
+         * @return updated request
+         */
+        public Request setPassword(String walletPassword) {
```
[the remark this answers is not in the document]
Comment (arjav-desai) [reply integrations/oci/atp/src/main/java/io/helidon/integrations/oci/atp/GenerateAutonomousDatabaseWalletRx.java:58]: Fixed the name.

File: examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java (hunk 2 above)
Comment (arjav-desai) [reply examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java]: Updated all names to "camel case".

File: examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java (hunk 3 above)
Comment (arjav-desai) [reply examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/ATPResource.java]: taken care with update to "camel case" naming.

**Review by tomas-langer — APPROVED** [review]
LGTM

**Pull request merged**